### PR TITLE
Move CLI to separate crate

### PR DESCRIPTION
**Pull request by jenrik**
Moves the commandline interface into a separate crate. This is step on the way to create alternative frontends, and for using the model checker as a library in other projects.

There haven't been made any meaningful decision on which APIs to make public, this is left as future work.

### CODE REVIEW

**Review by NicEastvillage — CHANGES_REQUESTED** [review]
The merge conflicts might suck for this one. Also:

File: cli/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+[package]
+name = "cli"
+version = "0.1.0"
+authors = ["Jener Rasmussen <jener@jener.dk>"]
```
[the remark this answers is not in the document]
Comment (jenrik) [reply cli/Cargo.toml:4]: I was waiting for #133 to go through

[the remark this answers is not in the document]
Comment (jenrik) [reply cli/Cargo.toml]: Fixed in 618d4b69af2fd5631cd5c2772813e740779a2e27

**Review by NicEastvillage — APPROVED** [review]

**Comment by cogitantium** [comment]
@jenrik tests are failing :eyes:

**Comment by cogitantium** [comment]
> @jenrik tests are failing eyes

Dang, they're failing on other branches with no changes. I'd wager a beer that it's test-env-log and tracing-subscriber being mad at eachother again.

**Review by cogitantium — APPROVED** [review]

**Pull request merged**